src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,313 @@
+
+        final List<String> actualNames = new ArrayList<>();
+        final List<String> actualSeq = new ArrayList<>();
+        extractNamesAndSeqFromFASTQ(actualFile, fastqFilesWellFormed, actualNames, actualSeq);
+
+        final List<String> expectedNames = new ArrayList<>();
+        final List<String> expectedSeq = new ArrayList<>();
+        extractNamesAndSeqFromFASTQ(expectedFile, fastqFilesWellFormed, expectedNames, expectedSeq);
+
+        Assert.assertEquals(expectedNames, actualNames);
+
+        // in stead of more stringent tests, e.g. "Assert.assertEquals(expectedSeq, actualSeq);",
+        // compute distance and let caller decide how large an edit distance is allowed
+        for(int i=0; i<expectedSeq.size(); ++i){
+            distances.add(StringUtils.getLevenshteinDistance(expectedSeq.get(i), actualSeq.get(i)));
+        }
+
+        //return RunSGAViaProcessBuilderOnSpark.extractBaseNameWithoutExtension(actualFile) + ".fa";
+        return FilenameUtils.getBaseName(actualFile.getName()) + ".fa";
+    }
+
+    // utility function: for extracting read names and sequences from a fastq/fasta file
+    private static void extractNamesAndSeqFromFASTQ(final File FASTAFile, final boolean fastqFilesWellFormed, List<String> readNames, List<String> sequences) throws IOException{
+
+        if(fastqFilesWellFormed){
+            try(final FastqReader reader = new FastqReader(FASTAFile)){
+                while(reader.hasNext()){
+                    final FastqRecord record = reader.next();
+                    readNames.add(record.getReadHeader());
+                    sequences.add(record.getReadString());
+                }
+            }
+        }else{
+            try(final BufferedReader reader = new BufferedReader(new FileReader(FASTAFile))){
+                String line = "";
+                int i=0;
+                while ((line = reader.readLine()) != null){
+                    final int l = i%4;
+                    if(0==l){
+                        readNames.add(line);
+                        ++i;
+                    }else{
+                        sequences.add(line);
+                        reader.readLine(); reader.readLine();
+                        i+=3;
+                    }
+                }
+            }
+        }
+    }
+
+    // utility function: generate mapping from contig length to its DNA sequence
+    // this is possible because test result contigs have unique length values
+    private static SortedMap<Integer, String> collectContigsByLength(final List<String> contigFileContents){
+
+        final Iterator<String> it = contigFileContents.iterator();
+
+        final SortedMap<Integer, String> result = new TreeMap<>();
+        int i=0;
+        while(it.hasNext()){
+            final String line = it.next();
+            if((++i)%2==1) {
```

Comment:
\[FindBugs\] INFO: IM: Check for oddness that won't work for negative numbers in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.collectContigsByLength\(List\)